jgscm/__init__.py

```diff
@@ -522,13 +523,15 @@ def _get_bucket(self, name, throw=False):
             cache[name] = bucket
             return bucket
 
-    @staticmethod
-    def _parse_path(path):
+    def _parse_path(self, path):
         """
         Splits the path into bucket name and path inside the bucket.
         :param path: string to split.
         :return: tuple(bucket name, bucket path).
         """
+        if self.default_path and not path.startswith(self.default_path):
```

Comment:
I removed the check for "/" prefix. At some point, if we decide we want to continue using this library, I think we can simplify the program, to stop stripping leading "/" in every function that also calls parse_path. Until then, it appears _parse_path is never called with a leading "/".

I can add that back in for safety.

Reply:
> Until then, it appears _parse_path is never called with a leading "/".

Ah, OK.  An assert to that effect might be good documentation.

One last potential bug: path might start with default_path, but it is just a prefix (e.g. default_path = foo, path = foobar/baz).  In general, this is never going to work if you create a file in the bucket with the same name as the bucket.